src/linker/Linker.Steps/SweepStep.cs

```diff
@@ -49,19 +50,31 @@ public SweepStep (bool sweepSymbols = true)
 
 		protected override void Process ()
 		{
-			assemblies = Context.Annotations.GetAssemblies ().ToArray ();
+			// To keep facades, scan all references so that even unused facades are kept
```

Comment:
This seems to be what https://github.com/mono/linker/blob/master/test/Mono.Linker.Tests.Cases/TypeForwarding/UnusedForwarderWithAssemblyLinkedAndFacadesKept.cs is checking. It is unusual though - the kept facade is empty (its forwarders are removed, it only has a module type). Assuming that is a bug, we can avoid loading references here.

Reply:
Get rid of this

Reply:
I'd like to scope this change to preserving the current facade behavior, while also enabling loading of dynamically referenced assemblies. Loading assemblies here is only necessary for `--keep-facades`, and it will go away when we fix https://github.com/mono/linker/issues/1740.